examples/bq_file_load_benchmark/benchmark_tools/file_generator.py

```diff
@@ -0,0 +1,662 @@
+                file should have. For exapmle, if the combination has
+                numFiles=10000, num_files should be 10000.
+        """
+
+        source_blob = self.bucket.get_blob(source_blob_name)
+        # Copy the source file until the combination has the correct number
+        # of files.
+        for n in range(start_num, num_files + 1):
+            file_string = 'file{0:d}'.format(n)
+            copied_destination = '{0:s}{1:s}.{2:s}'.format(
+                destination_path,
+                file_string,
+                extension,
+            )
+            copied_blob = self.bucket.copy_blob(
+                blob=source_blob,
+                destination_bucket=self.bucket,
+                new_name=copied_destination,
+            )
+            logging.info('Created file: {0:s}'.format(copied_blob.name))
+
+    def create_files(self):
+        """Creates all file combinations and store in GCS.
+
+        Generates list of file combination from parameters in
+        file_parameters.FILE_PARAMETERS dictionary, and creates each file in
+        the list, as long as it doesn't yet exist in self.bucket.
+        While each file is generated from a BigQuery staging table and stored
+        in GCS, the method of creating the file varies depending on the
+        parameters in the combination.
+        """
+        # Gather file parameters and constants.
+        files_consts = file_constants.FILE_CONSTANTS
+        file_types = self.file_params['fileType']
+        extract_formats = files_consts['extractFormats']
+        file_compression_types = self.file_params['fileCompressionTypes']
+        file_counts = self.file_params['numFiles']
+
+        # Begin the process of iterating through each combination.
+        logging.info('Starting to create files by exporting staging tables to '
+                     'bucket {0:s}'.format(self.bucket_name))
+        skip_message = 'Skipped path and its subsequent files: {0:s}'
+
+        # Gather a list of the staging tables. The staging tables already
+        # include the columnTypes, numColumns, and stagingDataSizes
+        # parameters (ex: the staging table 100_STRING_10_10MB has
+        # columnType=100_STRING, numColumns=10, and stagingDataSizes=10MB).
+        tables = self.primitive_staging_tables
+        for table_list_item in tables:
+            staging_table_util = table_util.TableUtil(
+                table_list_item.table_id,
+                table_list_item.dataset_id,
+            )
+            staging_table_util.set_table_properties()
+            # For each staging table, extract to each fileType, each
+            # compressionType, and copy each file so that the combination has
+            # the correct numFiles.
+            for file_type in file_types:
+                for compression_type in file_compression_types[file_type]:
+
+                    for num_files in file_counts:
```

Comment:
seems like 3 of these  4 nested for loops could be more readable using cartesian product of the lists you want to iterate over using 
```python
for (table_list_item, file_type, compression_type, num_files) 
    in itertools.product(tables, file_types, num_files):
    for compression_type in file_compression_types[file_type]:
        staging_table_util = table_util.TableUtil(
                table_list_item.table_id,
                table_list_item.dataset_id,
            )
            staging_table_util.set_table_properties()
            <everything in the fille_counts loop>